src/main/clojure/clara/rules/dsl.clj

```diff
@@ -135,7 +135,12 @@
            (parse-expression nested-expr expr-meta)))
 
    (contains? #{'test :test} (first expression))
-   {:constraints (vec (rest expression))}
+   (if (= 1 (count expression))
+     (throw-dsl-ex (str "Empty test conditions are not allowed.")
+                   {}
+                   expr-meta)
+     {:constraints (vec (rest expression))})
+
 
```

Comment:
Throw-dsl-ex won't include useful info here like what the offending rule or query is, but that looks like a shortcoming of the current usage of throw-dsl-ex in this namespace that impacts other malformed productions as well, so I think changing that is out of scope here.  I'll log another issue.

Reply:
For reference the issue logged is https://github.com/cerner/clara-rules/issues/351